### PR TITLE
Update master from dev

### PR DESCRIPTION
**Pull request by alejandrojnm**
- A lot of change
- Remove dependency from external lib to send form to the backend
- Now all POST, PUT, DELETE will send using Json format
- Add Reason in HTTPError

### CODE REVIEW

File: instance_test.go [hunk 1]
```diff
@@ -235,8 +248,8 @@ func TestDeleteInstance(t *testing.T) {
 
 func TestRebootInstance(t *testing.T) {
 	client, server, _ := NewAdvancedClientForTesting(map[string]map[string]string{
-		"/v2/instances/12345/hard_reboots": map[string]string{
-			"requestBody":  ``,
+		"/v2/instances/12345/hard_reboots": {
+			"requestBody":  `""`,
```
Comment (andyjeffries) [inline instance_test.go]: Still think we should replace the body if it's JSON with `{}` if it's `""`. Not sure Rails will handle an empty string as a JSON body.

Comment (alejandrojnm) [reply instance_test.go]: mm, i can try, let me check this

Comment (alejandrojnm) [reply instance_test.go]: @andyjeffries All good i try this:

`resp, err := client.RebootInstance("id_instances")`

and the return was:

`&{success   }`

Comment (andyjeffries) [reply instance_test.go]: OK then :)

**Comment by andyjeffries** [comment]
"This branch has conflicts that must be resolved"

**Comment by alejandrojnm** [comment]
@andyjeffries all is ready to make the merge

**Comment by andyjeffries** [comment]
Can you squash the commits, give a single commit message that reflects what it is and then fast-forward merge it (so we keep a clean history), rather than have commits like "Fix error in the merge" and "Update master from dev". https://medium.com/@catalinaturlea/clean-git-history-a-step-by-step-guide-eefc0ad8696d https://christoph-rumpel.com/2015/05/clean-up-your-commits-for-a-pull-request

**Pull request merged**